artifactory/commands/jobsummariesimpl/buildpublishjobsummary.go

```diff
@@ -0,0 +1,77 @@
+package jobsummariesimpl
+
+import (
+	"encoding/json"
+	"fmt"
+	buildInfo "github.com/jfrog/build-info-go/entities"
+	"github.com/jfrog/jfrog-cli-core/v2/jobsummaries"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"strings"
+	"time"
+)
+
+type GithubSummaryBpImpl struct {
+	Builds []*buildInfo.BuildInfo
+}
+
+func (ga *GithubSummaryBpImpl) CreateSummaryMarkdown(content any, section jobsummaries.MarkdownSection) (err error) {
+	return jobsummaries.CreateSummaryMarkdownBaseImpl(content, section, ga.appendResultObject, ga.renderContentToMarkdown)
+}
+
+func (ga *GithubSummaryBpImpl) appendResultObject(currentResult interface{}, previousResults []byte) ([]byte, error) {
+	build, ok := currentResult.(*buildInfo.BuildInfo)
+	if !ok {
+		return nil, fmt.Errorf("failed to cast currentResult to buildInfo.BuildInfo")
+	}
+	// Unmarshal the data into an array of build info objects
+	var builds []*buildInfo.BuildInfo
+	if len(previousResults) > 0 {
+		err := json.Unmarshal(previousResults, &builds)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Append the new build info object to the array
+	builds = append(builds, build)
+	return json.Marshal(builds)
+}
+
+func (ga *GithubSummaryBpImpl) renderContentToMarkdown(content []byte) (markdown string, err error) {
```

Comment:
renderContentToMarkdown --> toMarkdown

Let's create a unit test for this function.